proto/p4/config/v1/p4info.proto

```diff
@@ -48,6 +48,19 @@ message Documentation {
   string description = 2;
 }
 
+// Used to describe the required properties of the underlying platform.
+message PlatformProperties {
```

Comment:
@jafingerhut, @chrispsommers, and @smolkaj, I wrapped this into a PlatformProperties message as desired.

I'm thinking best to add it to the capabilities in a separate PR because it will need a totally separate description anyway. WDYT?